torch/_subclasses/fake_tensor.py

```diff
@@ -1232,6 +1271,14 @@ def dispatch(self, func, types, args=(), kwargs=None):
             # NOTE: incr is intentionally unused for a RAII pattern
             incr = IncrementRecursionCount()
 
+        # NOTE: Skip fakifying inputs to torch.ops.profiler because they only have c++ implementation
+        # and are performance critical. The suggested workaround will cause non-eligible overhead.
+        if func not in {
+            torch.ops.profiler._record_function_enter_new,
+            torch.ops.profiler._record_function_exit._RecordFunction,
+        }:
+            args, kwargs = _maybe_fakify_script_object(args, kwargs)
```

Comment:
How did you run into a case where we are fakifying inputs to torch.ops.profiler?

Reply:
_record_function_enter_new, _record_function_exit._RecordFunction defined one implementation for all [backends](https://github.com/pytorch/pytorch/blob/main/torch/csrc/autograd/record_function_ops.cpp#L129-L140).

[dynamo_timed](https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/utils.py#L226-L273) used them in inductor to profile the compilation time and it's on the critical path. [Test failure of triggering the fakification](https://hud.pytorch.org/pytorch/pytorch/pull/120045?sha=caa2d677aa22c37591f5d9d9edbaaf797314292b) 
```
2024-02-16T01:36:43.4777090Z     torch.ScriptObject, lambda x: _wrap_script_object(x), (args, kwargs)
2024-02-16T01:36:43.4778229Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 1200, in _wrap_script_object
2024-02-16T01:36:43.4779290Z     fake_class = torch.library.registered_class[x._type().qualified_name()]
2024-02-16T01:36:43.4780112Z torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
2024-02-16T01:36:43.4780857Z KeyError: '__torch__.torch.classes.profiler._RecordFunction'
```